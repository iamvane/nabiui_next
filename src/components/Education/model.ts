
export interface EducationType {
  id?: number;
  school: string;
  graduationYear: string;
  degreeType: string;
  fieldOfStudy: string;
  schoolLocation: string;
}
