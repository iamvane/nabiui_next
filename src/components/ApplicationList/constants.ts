
export namespace ApplicationListComponent {
  export const breadcrumbLabels = {
    home: 'Home',
    applications: 'Applications'
  };

  export const pageTitle = 'Applications';
}


export namespace ApplicationCardComponent {
  export const availabilitySection = 'Availability';
  export const messageSection = 'Message';
  export const rate = 'Rate:'
  export const bookLessons = 'Book Lessons'
}
