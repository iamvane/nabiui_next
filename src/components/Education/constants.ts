export namespace EducationComponent {
  export enum Text {
    YourEducation = 'Your Education',
    TellStudentsAbout = 'Tell students about your educational background',
    AddEducation = 'Add Education'
  }
}

export namespace EducationFormComponent {
  export const extraYears = 7;

  export enum Text {
    AddButton = 'Add',
    SaveChangesButton = 'Save Changes',
    CancelButton = 'Cancel'
  }

  export enum FieldNames {
    School = 'school',
    GraduationYear = 'graduationYear',
    FieldOfStudy = 'fieldOfStudy',
    DegreeType = 'degreeType',
    SchoolLocation = 'schoolLocation'
  }

  export enum Ids {
    School = 'school',
    GraduationYear = 'graduationYear',
    FieldOfStudy = 'fieldOfStudy',
    DegreeType = 'degreeType',
    SchoolLocation = 'schoolLocation'
  }

  export enum Placeholders {
    School = 'School Name',
    FieldOfStudy = 'Field of study (major or concentration)',
    SchoolLocation = 'School location (city, state, country)'
  }

  export enum DisabledPlaceholders {
    GraduationYear= 'Graduation year',
    Degreetype = 'Degree type'
  }
}

export const degreeTypeKey = {
  associate: 'associate',
  bachelors: 'bachelors',
  graduate: 'graduate',
  professional: 'professional',
  certification: 'certification',
  other: 'other'
};

export const degreeTypeLabels = {
  [degreeTypeKey.associate]: 'Associate Degree',
  [degreeTypeKey.bachelors]: 'Bachelor\'s Degree',
  [degreeTypeKey.graduate]: 'Graduate Degree',
  [degreeTypeKey.professional]: 'Professional Degree',
  [degreeTypeKey.certification]: 'Certification',
  [degreeTypeKey.other]: 'Other'
};

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degree',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface Degree {
  value: string;
  label: DegreeType;
}

export const degreeOptions = {
  Associate: <Degree> {
    value: degreeTypeKey.associate,
    label: degreeTypeLabels.associate
  },
  Bachelor: <Degree> {
    value: degreeTypeKey.bachelors,
    label: degreeTypeLabels.bachelors
  },
  graduate: <Degree> {
    value: degreeTypeKey.graduate,
    label: degreeTypeLabels.graduate
  },
  Professional: <Degree> {
    value: degreeTypeKey.professional,
    label: degreeTypeLabels.professional
  },
  Certification: <Degree> {
    value: degreeTypeKey.certification,
    label: degreeTypeLabels.certification
  },
  Other: <Degree> {
    value: degreeTypeKey.other,
    label: degreeTypeLabels.other
  }
};

export namespace EducationAddedComponent {
  export const concentration = '{degreeTypeReplace}, {fieldOfStudyReplace}';
  export const degreeTypePlaceholder = '{degreeTypeReplace}';
  export const fieldOfStudyPlaceholder = '{fieldOfStudyReplace}';
}
