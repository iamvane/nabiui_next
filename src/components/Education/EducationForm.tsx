import * as React from 'react';

import {
  Button,
  FormControl,
  Grid,
  Input,
  Select,
  TextField
} from '@material-ui/core';
import Save from '@material-ui/icons/Save';
import Add from '@material-ui/icons/Add';
import Close from '@material-ui/icons/Close';

import { YearsList } from '../../utils/YearsList';
import { EducationType } from './model';
import { EducationFormComponent, degreeOptions } from './constants';

interface Props extends EducationType {
  handleChange: (event: React.FormEvent<{}>) => void;
  handleOnBlur: (event: React.FormEvent<{}>) => void;
  handleSave: (event: React.FormEvent<{}>) => void;
  handleCancel: () => void;
  allFieldsFilled: boolean;
  isEditing: boolean;
}

const EducationForm: React.StatelessComponent<Props> = props => {
  const selectOptions: any = [];

  for (const [key, value] of Object.entries(degreeOptions)) {
    selectOptions.push(<option key={key} value={value.value}>{value.label}</option>);
  }

  let years = YearsList(EducationFormComponent.extraYears);

  const yearsItems = years.map((year) =>
    <option key={year.toString()} value={year}>{year}</option>
  );

  const editSubmitButton: JSX.Element = (
    <Button
      color="primary"
      variant="contained"
      className="nabi-margin-top-small nabi-text-uppercase"
      onClick={props.handleSave}
      type="submit"
      disabled={!props.allFieldsFilled ? true : false}
    >
      <Save className="nabi-margin-right-xsmall" />
      <span className="nabi-margin-left-xsmall">{EducationFormComponent.Text.SaveChangesButton}</span>
    </Button>
  );

  const addSubmitButton: JSX.Element = (
    <Button
      color="primary"
      variant="contained"
      className="nabi-margin-top-small nabi-text-uppercase"
      onClick={props.handleSave}
      type="submit"
      disabled={!props.allFieldsFilled ? true : false}
    >
      <Add className="nabi-margin-right-xsmall" />
      <span className="nabi-margin-left-xsmall">{EducationFormComponent.Text.AddButton}</span>
    </Button>
  );

  return (
    <Grid container={true}>
      <Grid item={true} md={8} xs={12} sm={10}>
        <form>
          <TextField
            onChange={props.handleChange}
            onBlur={props.handleOnBlur}
            fullWidth={true}
            id={EducationFormComponent.Ids.School}
            margin="normal"
            name={EducationFormComponent.FieldNames.School}
            placeholder={EducationFormComponent.Placeholders.School}
            required={true}
            value={props.school}
          />
          <FormControl
            fullWidth={true}
          >
            <Select
              native={true}
              input={
                <Input
                  id={EducationFormComponent.Ids.GraduationYear}
                  name={EducationFormComponent.FieldNames.GraduationYear}
                />
              }
              value={props.graduationYear}
              onChange={props.handleChange}
              onBlur={props.handleOnBlur}
              style={{
                border: '1px solid red'
              }}
            >
              <option value="" disabled={true}>
                {EducationFormComponent.DisabledPlaceholders.GraduationYear}
              </option>
              {yearsItems}
            </Select>
          </FormControl>
          <TextField
            onChange={props.handleChange}
            onBlur={props.handleOnBlur}
            fullWidth={true}
            id={EducationFormComponent.Ids.FieldOfStudy}
            margin="normal"
            name={EducationFormComponent.FieldNames.FieldOfStudy}
            placeholder={EducationFormComponent.Placeholders.FieldOfStudy}
            required={true}
            value={props.fieldOfStudy}
          />
          <FormControl
            fullWidth={true}
          >
            <Select
              native={true}
              input={
                <Input
                  id={EducationFormComponent.Ids.DegreeType}
                  name={EducationFormComponent.FieldNames.DegreeType}
                />
              }
              value={props.degreeType}
              onChange={props.handleChange}
              onBlur={props.handleOnBlur}
              variant="outlined"
            >
              <option value="" disabled={true}>{EducationFormComponent.DisabledPlaceholders.Degreetype}</option>
              {selectOptions}
            </Select>
          </FormControl>
          <TextField
            onChange={props.handleChange}
            onBlur={props.handleOnBlur}
            fullWidth={true}
            id={EducationFormComponent.Ids.SchoolLocation}
            margin="normal"
            name={EducationFormComponent.FieldNames.SchoolLocation}
            placeholder={EducationFormComponent.Placeholders.SchoolLocation}
            required={true}
            value={props.schoolLocation}
          />
          {props.isEditing ? editSubmitButton : addSubmitButton}
          <Button
            color="default"
            variant="contained"
            className="nabi-margin-top-small nabi-text-uppercase nabi-margin-left-xsmall"
            onClick={props.handleCancel}
          >
            <Close className="nabi-margin-right-xsmall" />
            <span className="nabi-margin-left-xsmall">{EducationFormComponent.Text.CancelButton}</span>
          </Button>
        </form>
      </Grid>
    </Grid>
  );
};

export default EducationForm;
