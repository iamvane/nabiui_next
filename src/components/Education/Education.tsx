import React from 'react';
import { useState, useEffect, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  bindActionCreators,
} from 'redux';

import {
  Button,
  CircularProgress,
  Typography,
} from '@material-ui/core';
import ArrowForward from '@material-ui/icons/ArrowForward';
import Add from '@material-ui/icons/Add';

import { StoreState } from '../../redux/reducers/store';
import {
  fetchEducation,
  addEducation,
  editEducation,
  deleteEducation
} from '../../redux/actions/InstructorActions';
import { Routes } from '../common/constants/Routes';
import { ProfileBuilderStepper } from '../ProfileBuilder/constants';
import { StepperButtons } from '../CommonStepper/StepperButtons';
import SectionTitle from '../common/SectionTitle';
import SnackBar from '../common/SnackBar';
import EducationForm from './EducationForm';
import EducationAdded from './EducationAdded';
import { EducationType } from './model';
import { EducationComponent } from './constants';

const Education = () => {
  const [educationId, setEducationId] = useState(undefined);
  const [school, setSchool] = useState('');
  const [graduationYear, setGraduationYear] = useState('');
  const [schoolLocation, setSchoolLocation] = useState('');
  const [degreeType, setDegreeType] = useState('');
  const [fieldOfStudy, setFieldOfStudy] = useState('');
  const [educationFormIsOpen, showEducationForm] = useState(false);
  const [isEditing, setIsEditing] = useState(false);
  const [allFieldsFilled, setAllFieldsFilled] = useState(false);
  const [education, setEducation] = useState([]);
  const [snackbarIsOpen, showSnackbar] = useState(false);
  const [snackbarMessage, setSnackbarMessage] = useState('');
  const [snackbarMessageType, setSnackbarMessageType] = useState('success');

  const closeSnackbar = () => showSnackbar(false);

  const dispatch = useDispatch();
  const fetchEducationAction = bindActionCreators(fetchEducation, dispatch);
  const addEducationAction = bindActionCreators(addEducation, dispatch);
  const editEducationAction = bindActionCreators(editEducation, dispatch);
  const deleteEducationAction = bindActionCreators(deleteEducation, dispatch);

  const resetEducationState = () => {
    setSchool('');
    setGraduationYear('');
    setSchoolLocation('');
    setDegreeType('');
    setFieldOfStudy('');
  }
  const toggleEducationForm = () => {
    showEducationForm(!educationFormIsOpen);
  }

  const handleCancel = () => {
    resetEducationState();
    toggleEducationForm();
    setIsEditing(false);
  }

  const deleteEducationAsync = (educationId: number): void => {
    deleteEducationAction(educationId);
  }

  let {
    instructorEducation,
    isAddEducationRequesting,
    isDeleteEducationRequesting,
    isEditEducationRequesting,
    isFetchEducationRequesting,
    addEducationError,
    fetchEducationError,
    editEducationError,
    deleteEducationError,
  } = useSelector((state: StoreState) => {
    const {
      instructor: {
        education: instructorEducation
      },
      actions: {
        fetchEducation: {
          isRequesting: isFetchEducationRequesting,
          error: fetchEducationError
        },
        addEducation: {
          isRequesting: isAddEducationRequesting,
          error: addEducationError
        },
        editEducation: {
          isRequesting: isEditEducationRequesting,
          error: editEducationError,
        },
        deleteEducation: {
          isRequesting: isDeleteEducationRequesting,
          error: deleteEducationError,
        }
      }
    } = state.instructor;

    return {
      isFetchEducationRequesting,
      isAddEducationRequesting,
      isDeleteEducationRequesting,
      isEditEducationRequesting,
      fetchEducationError,
      addEducationError,
      editEducationError,
      deleteEducationError,
      instructorEducation: instructorEducation || [],
    };
  });

  useEffect(() => {
    if (!instructorEducation.length) {
      fetchEducationAction();
    }
    if (instructorEducation.length) {
      setEducation(instructorEducation);
      showEducationForm(false);
      setIsEditing(false);
    } else {
      setEducation([]);
      showEducationForm(false);
      setIsEditing(false);
    }
  },
    [
      JSON.stringify(instructorEducation)
    ]);


  useEffect(() => {
    if (instructorEducation.length > education.length) {
      setEducation(instructorEducation);
      showEducationForm(false);
      setIsEditing(false);
    }
  },
    [
      JSON.stringify(instructorEducation)
    ]);

  useEffect(() => {
    if (addEducationError) {
      setSnackbarMessage(addEducationError);
      setSnackbarMessageType('error');
      showSnackbar(true);
      setIsEditing(false);
    }
  },
    [
      addEducationError
    ]);


  useEffect(() => {
    if (editEducationError) {
      setSnackbarMessage(editEducationError);
      setSnackbarMessageType('error');
      showSnackbar(true);
      setIsEditing(false);
    }
  },
    [
      editEducationError
    ]);

  useEffect(() => {
    if (deleteEducationError) {
      setSnackbarMessage(deleteEducationError);
      setSnackbarMessageType('error');
      showSnackbar(true);
      setIsEditing(false);
    }
  },
    [
      deleteEducationError
    ]);


  const handleChange = useCallback(
    (event: React.FormEvent<HTMLInputElement>): void => {
      const target = event.currentTarget;
      let value = target.value as any;
      let name = target.name;

      if (name === 'school') {
        setSchool(value);
      }
      if (name === 'graduationYear') {
        setGraduationYear(value);
      }
      if (name === 'schoolLocation') {
        setSchoolLocation(value);
      }
      if (name === 'degreeType') {
        setDegreeType(value);
      }
      if (name === 'fieldOfStudy') {
        setFieldOfStudy(value);
      }
    },
    []
  );

  useEffect(() => {
    const allFieldsFilled = [
      school,
      graduationYear,
      schoolLocation,
      degreeType,
      fieldOfStudy
    ].every((field) => {
      if (typeof field === 'number') {
        return field > 0;
      }
      return field.length;
    });

    setAllFieldsFilled(allFieldsFilled);
  }, [
    school,
    graduationYear,
    schoolLocation,
    degreeType,
    fieldOfStudy
  ]);

  const editEducationForm = useCallback(
    (educationId: number) => {
      const educationToEdit = instructorEducation.length && instructorEducation.find(education =>
        education.id === educationId
      );
      if (educationToEdit) {
        setIsEditing(true);
        setEducationId(educationToEdit.id);
        setSchool(educationToEdit.school);
        setGraduationYear(educationToEdit.graduationYear);
        setSchoolLocation(educationToEdit.schoolLocation);
        setDegreeType(educationToEdit.degreeType);
        setFieldOfStudy(educationToEdit.fieldOfStudy);

        showEducationForm(true);
      }
    },
    [
      educationFormIsOpen,
      isEditing,
      school,
      graduationYear,
      schoolLocation,
      degreeType,
      fieldOfStudy,
      educationId,
      JSON.stringify(instructorEducation)
    ]
  );

  const handleSave = useCallback(
    () => {
      const education: EducationType = {
        school,
        graduationYear,
        degreeType,
        fieldOfStudy,
        schoolLocation
      };

      console.log('is editing', true);
      if (isEditing) {
        education.id = educationId;
        editEducationAction(education);
      } else {
        addEducationAction(education);
      }
      resetEducationState();
      toggleEducationForm();
    },
    [
      school,
      graduationYear,
      degreeType,
      fieldOfStudy,
      schoolLocation,
      educationId
    ]
  );

  const requesting = isAddEducationRequesting || isEditEducationRequesting ||
    isFetchEducationRequesting || isDeleteEducationRequesting;

  const renderEducationForm = (): JSX.Element => {
    return (
      <div>
        <Typography className="nabi-margin-top-xsmall nabi-text-uppercase" variant="body2">
          {EducationComponent.Text.AddEducation}
        </Typography>
        <EducationForm
          handleChange={handleChange}
          handleSave={handleSave}
          school={school}
          graduationYear={graduationYear}
          degreeType={degreeType}
          fieldOfStudy={fieldOfStudy}
          schoolLocation={schoolLocation}
          handleCancel={handleCancel}
          isEditing={isEditing}
          allFieldsFilled={allFieldsFilled}
        />
      </div>
    );
  }

  const educationAdded = education.map((education, i) => (
    <li className="nabi-list" key={i}>
      <EducationAdded
        id={education.id}
        gridWidth={6}
        school={education.school}
        graduationYear={education.graduationYear}
        schoolLocation={education.schoolLocation}
        degreeType={education.degreeType}
        fieldOfStudy={education.fieldOfStudy}
        deleteEducation={(educationId: number) => deleteEducationAsync(educationId)}
        editEducation={(educationId: number) => editEducationForm(educationId)}
      />
    </li>
  ));

  return (
    <div>
      {requesting ?
        <div className="nabi-text-center">
          <CircularProgress />
        </div> :
        (!educationFormIsOpen ?
          <div>
            <SectionTitle text={EducationComponent.Text.YourEducation} />
            <Typography className="nabi-margin-top-xsmall">
              {EducationComponent.Text.TellStudentsAbout}
            </Typography>
            <ul>
              {educationAdded}
            </ul>
            <div className="nabi-margin-top-medium">
              <Button
                color="primary"
                variant="contained"
                onClick={toggleEducationForm}
              >
                <Add className="nabi-margin-right-xsmall" />
                {EducationComponent.Text.AddEducation}
              </Button>
            </div>
          </div>
          : renderEducationForm())
      }
      <StepperButtons
        nextPath={Routes.BuildProfile + ProfileBuilderStepper.StepsPaths.Employment}
        backPath={Routes.BuildProfile + ProfileBuilderStepper.StepsPaths.JobPreferences}
        icon={<ArrowForward />}
        errors={
          fetchEducationError ||
          addEducationError ||
          editEducationError ||
          deleteEducationError
        }
      />
      <SnackBar
        isOpen={snackbarIsOpen}
        message={snackbarMessage}
        handleClose={closeSnackbar}
        variant={snackbarMessageType}
        hideIcon={true}
      />
    </div>
  );
}

export default Education;
