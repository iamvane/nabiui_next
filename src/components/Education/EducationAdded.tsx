import * as React from 'react';

import {
  Icon,
  IconButton,
  Grid,
  Typography
} from '@material-ui/core';

import { EducationType }  from 'components/Education/model';
import {
  degreeTypeLabels,
  EducationAddedComponent
} from 'components/Education/constants';

interface Props extends EducationType {
  deleteEducation: (educationId: number | undefined) => void;
  editEducation: (educationId: number | undefined) => void;
  gridWidth?: boolean | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | undefined;
  notEditable?: boolean;
}

const EducationAdded: React.StatelessComponent<Props> = props => {
  return (
    <Grid
      item={true}
      md={props.gridWidth && props.gridWidth}
      xs={12}
      className={`${props.notEditable ? '' : 'nabi-editable-item'} nabi-margin-top-small`}
    >
      <div className="nabi-action-buttons">
        <IconButton
          color="primary"
          className="nabi-float-right nabi-margin-left-xsmall"
          aria-label="Delete"
          onClick={() => props.deleteEducation(props.id)}
        >
          <Icon>delete</Icon>
        </IconButton>

        <IconButton
          color="primary"
          className="nabi-float-right"
          aria-label="Edit"
          onClick={() => props.editEducation(props.id)}
        >
          <Icon>edit</Icon>
        </IconButton>
      </div>

      <Typography className="nabi-text-uppercase" variant="body2">
        <mark>{props.school}</mark>
      </Typography>

      <Typography className="nabi-margin-top-xsmall">
        <mark>
          {EducationAddedComponent.concentration.replace(
            EducationAddedComponent.degreeTypePlaceholder,
            degreeTypeLabels[props.degreeType]
          ).replace(
            EducationAddedComponent.fieldOfStudyPlaceholder,
            props.fieldOfStudy
          )}
        </mark>
      </Typography>

      <Typography>
        <mark>{props.graduationYear}</mark>
      </Typography>

      <Typography>
        <mark>{props.schoolLocation}</mark>
      </Typography>
    </Grid>
  );
};

export default EducationAdded;
