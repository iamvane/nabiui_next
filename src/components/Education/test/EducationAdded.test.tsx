import * as React from 'react';
import { shallow } from 'enzyme';
import EducationAdded from 'components/Education/EducationAdded';
import { DegreeType } from 'components/Education/constants';

describe('EducationAdded', () => {
  let wrapper: any;
  const deleteEducation: () => void = jest.fn();
  const editEducation: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <EducationAdded
          editEducation={editEducation}
          deleteEducation={deleteEducation}
          school=""
          graduationYear=""
          degreeType={DegreeType.bachelors}
          fieldOfStudy=""
          schoolLocation=""
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
