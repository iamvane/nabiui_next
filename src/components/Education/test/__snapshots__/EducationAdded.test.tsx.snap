// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationAdded Shallow Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EducationAdded
    degreeType="Bachelor's Degree"
    deleteEducation={[MockFunction]}
    editEducation={[MockFunction]}
    fieldOfStudy=""
    graduationYear=""
    school=""
    schoolLocation=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="nabi-action-buttons"
        >
          <WithStyles(IconButton)
            aria-label="Delete"
            className="nabi-float-right nabi-margin-left-xsmall"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              delete
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
          <WithStyles(IconButton)
            aria-label="Edit"
            className="nabi-float-right"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              edit
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
        </div>,
        <WithStyles(Typography)
          className="nabi-text-uppercase"
          variant="body2"
        >
          <mark>
            
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)
          className="nabi-margin-top-xsmall"
        >
          <mark>
            undefined, 
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            
          </mark>
        </WithStyles(Typography)>,
      ],
      "className": "nabi-editable-item nabi-margin-top-small",
      "item": true,
      "md": undefined,
      "xs": 12,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
          ],
          "className": "nabi-action-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>,
              "className": "nabi-float-right nabi-margin-left-xsmall",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "delete",
              },
              "ref": null,
              "rendered": "delete",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Edit",
              "children": <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>,
              "className": "nabi-float-right",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "edit",
              },
              "ref": null,
              "rendered": "edit",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            
          </mark>,
          "className": "nabi-text-uppercase",
          "variant": "body2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            undefined, 
          </mark>,
          "className": "nabi-margin-top-xsmall",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "undefined, ",
          },
          "ref": null,
          "rendered": "undefined, ",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "mark",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="nabi-action-buttons"
          >
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </div>,
          <WithStyles(Typography)
            className="nabi-text-uppercase"
            variant="body2"
          >
            <mark>
              
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall"
          >
            <mark>
              undefined, 
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              
            </mark>
          </WithStyles(Typography)>,
        ],
        "className": "nabi-editable-item nabi-margin-top-small",
        "item": true,
        "md": undefined,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Delete"
                className="nabi-float-right nabi-margin-left-xsmall"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
              <WithStyles(IconButton)
                aria-label="Edit"
                className="nabi-float-right"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
            ],
            "className": "nabi-action-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>,
                "className": "nabi-float-right nabi-margin-left-xsmall",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "delete",
                },
                "ref": null,
                "rendered": "delete",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Edit",
                "children": <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>,
                "className": "nabi-float-right",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "edit",
                },
                "ref": null,
                "rendered": "edit",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              
            </mark>,
            "className": "nabi-text-uppercase",
            "variant": "body2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              undefined, 
            </mark>,
            "className": "nabi-margin-top-xsmall",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined, ",
            },
            "ref": null,
            "rendered": "undefined, ",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "mark",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
